x-pack/plugins/spaces/public/management/edit_space/delete_spaces_button.tsx

```diff
@@ -42,7 +42,7 @@ export class DeleteSpacesButton extends Component<Props, State> {
 
     let ButtonComponent: any = EuiButton;
 
-    const extraProps: EuiButtonIconProps = {};
+    const extraProps: Partial<EuiButtonIconProps> = {};
```

Comment:
This change isn't type-safe -- if other required `EuiButtonIcon` properties are added in the future, the type checker wouldn't catch it.

Would you mind updating it to this?

```diff
diff --git a/x-pack/plugins/spaces/public/management/edit_space/delete_spaces_button.tsx b/x-pack/plugins/spaces/public/management/edit_space/delete_spaces_button.tsx
index cf131b4c640..a03d42d10c3 100644
--- a/x-pack/plugins/spaces/public/management/edit_space/delete_spaces_button.tsx
+++ b/x-pack/plugins/spaces/public/management/edit_space/delete_spaces_button.tsx
@@ -42,11 +42,11 @@ export class DeleteSpacesButton extends Component<Props, State> {
 
     let ButtonComponent: any = EuiButton;
 
-    const extraProps: Partial<EuiButtonIconProps> = {};
+    let extraProps: EuiButtonIconProps | undefined;
 
     if (this.props.style === 'icon') {
       ButtonComponent = EuiButtonIcon;
-      extraProps.iconType = 'trash';
+      extraProps = { iconType: 'trash' };
     }
 
     return (
```

Tested locally and it works =)

Reply:
Pushed your suggested change, thanks!